ethcore/src/client/client.rs

```diff
@@ -304,6 +304,13 @@ impl Importer {
 				// LockedBlock. See https://github.com/openethereum/openethereum/issues/11603
 				let preverified_header = block.header.clone();
 				let hash = block.header.hash();
+				if let Some(sync_until_block_nr) = client.config.sync_until {
+					if block.header.number() > sync_until_block_nr {
+						info!("Sync target reached at block: #{}. Going offline.", sync_until_block_nr);
+						client.disable();
+						break;
+					}
+				}
```

Comment:
I'm not sure this is the right place for the purpose, what if `block` is invalid? Maybe `commit_block` is a better place?

Reply:
I guess in case of reorgs it could be unfortunate to sync to the wrong chain, but it shouldn't matter for bench/testing purposes.

Reply:
I'm probably mentally biased by what I'd use this for but I don't see a use case outside bench/testing/debugging. For those cases reorgs don't matter much (or might even be a feature?).
What problems do you see with a vanilla user ending up on a bad fork be? When they start back up again they'd be reorg'd to the right chain I think? Or…?

Reply:
> I'm not sure this is the right place for the purpose, what if `block` is invalid? Maybe `commit_block` is a better place?

I tried a few different places for this. `commit_block()` is a good idea but the return type is `ImportRoute` (not `Result`) so I'd have to re-work that a bit more to be able to detect that we're going to sleep and short cut the for-loop in `import_verified_blocks()`. I also looked at putting it higher up the call stack but verification is multi threaded and it seemed less obvious what we were doing. I can have another stab at it if you feel this is the wrong place.

Reply:
Ignore my comment about reorgs, as I wrote, it shouldn't matter for bench purposes.
> what if block is invalid?

This still could be an issue, namely, how sure are use that `--sync-until=x` actually imports `x - 1` and doesn't stop earlier? I see the problem with `commit_block`, maybe then just move this check in `import_verified_blocks` but right before `commit_block`?

Reply:
> This still could be an issue, namely, how sure are you that `--sync-until=x` actually imports `x - 1` and doesn't stop earlier?

Moved the check right after `check_and_lock_block()` but I'm not entirely convinced it's right: if `x-n` for some `n` happens to be an invalid block we'll halt anyway and I think that's unsurprising (a Good Thing™). I think it's fair to assume that all blocks up to `x` are valid and if they are not, then it's not the fault of `--sync-until`?
Either way, it's not wrong to let `check_and_lock_block()` run before halting so… :)